coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -319,7 +319,12 @@ public boolean restore_state(InputObjectState os, int ot) {
                     }
 
                     if (status == LRAStatus.Active) {
-                        status = LRAStatus.Cancelling; // transition from Active to Cancelling
+                        if (getSize(pendingList) == 0 && getSize(preparedList) == 0 && getSize(heuristicList) == 0) {
+                            updateState(LRAStatus.Cancelled);
+                        }
+                        else {
+                            updateState(LRAStatus.Cancelling);
```

Comment:
This `updateState(LRAStatus.Cancelling);` is interesting and it could be a separate JBTM to explain why not saving the status before (the now called deactivate() in updateState would do that now I think) is determined to need to be changed now. I am also not sure if restore_state is normally a method that we expect to issue calls to save_state (via deactivate() in this case)?

Reply:
Good point, I'll create a JBTM for this PR and mention it here.

Reply:
https://issues.redhat.com/browse/JBTM-3971

Reply:
Thank you